File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        
        res,max_num = 0,0
        for i in range(len(colors)):
            if(i>0 and (colors[i] is not colors[i-1])):
                max_num = 0
            res += min (max_num,neededTime[i])
            max_num = max(max_num,neededTime[i])
        return res
            

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string colors, vector<int>& neededTime) {
        int sum = 0,max_num=0;
        for(int i=0;i<colors.size();i++){
            if(i>0 && colors[i]!=colors[i-1]){
                max_num = 0;
            }
            sum += min(max_num,neededTime[i]);
            max_num = max(max_num,neededTime[i]);
        }
        return sum;
    }
};